lib/IRGen/GenClangDecl.cpp

```diff
@@ -120,6 +216,29 @@ void IRGenModule::emitClangDecl(const clang::Decl *decl) {
     stack.push_back(D);
   });
 
+  std::vector<const clang::Decl *> ctors;
+  llvm::copy_if(stack, std::back_inserter(ctors), [](const clang::Decl *decl) {
+    return isa<clang::CXXConstructorDecl>(decl);
+  });
+
+  std::set<const clang::Decl *> visited;
+  std::vector<const clang::Decl *> functionDecls;
+  for (auto *ctor : ctors)
+    walkCallGraphFromCtor(cast<clang::CXXConstructorDecl>(ctor), visited,
+                          functionDecls);
+  llvm::sort(functionDecls);
```

Comment:
Does this sort add anything? Is it just for performance?

Reply:
I was thinking that a sort+unique would be better than a std::set for perf but I can try a std::set or one of the LLVM ADT sets.